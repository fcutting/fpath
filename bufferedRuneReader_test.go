package fpath

import "testing"

func Test_bufferedRuneReader_Get(t *testing.T) {
	input := "hello world"
	expectedRunes := []rune{'h', 'e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd'}
	brr := newBufferedRuneReader(input)

	for i, expected := range expectedRunes {
		r := brr.Get()
		if r != expected {
			t.Fatalf("Unexpected rune at position %d\nExpected: %q\n Actual: %q", i, expected, r)
		}
	}
}
