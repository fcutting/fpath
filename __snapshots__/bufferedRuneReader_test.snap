
[Test_tokenReader_getToken_InvalidRune - 1]
Invalid rune '`'
---
